### PR TITLE
[pages][tools]: Implement Tooling Analytics and Integrate Bowtie Report

### LINKED ISSUE #778: ✨ Enhancement: Implement Tooling Trends Analytics
**Description:**

Post user interactions and engagement patterns and analytics on the tooling page to identify trends and insights for further improvements.

**Major Tasks:**
- [x] **Analytics Integration**
  - Integrate Google Sheets Analytics for user interactions and traffic.
  - Set up custom events to monitor specific actions (e.g., filter usage, link clicks).

### PR DESCRIPTION
**Pull request by DarhkVoyd**
**What kind of change does this PR introduce?**
This PR aims to provide some bug fixes, clean code refactor, enhanced code and add new enhancements such as implementation of analytics for tooling trends for the tooling page and integrate bowtie report in the current design.

**Issue Number:**
-  Closes #778 #780 
-  Related to: #716  
-  Others? 

**Screenshots/videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

### CODE REVIEW

File: data/tooling-data.yaml [hunk 1]
```diff
@@ -901,6 +901,8 @@
   supportedDialects:
     draft: [6, 7, 2019-09, 2020-12]
   toolingListingNotes: 'Powered by JsonSchema.Net in Blazor WASM for client-side validation'
+  landscape:
```
Comment (benjagm) [inline data/tooling-data.yaml:904]: Why we have a parent property called Landscape? I think the logo is an important property and we don't need to group them into "landscape". I suggest avoiding it and creating the logo as any other property.

File: next.config.js [hunk 2]
```diff
@@ -5,7 +5,31 @@ const nextConfig = {
   pageExtensions: ['page.tsx'],
   images: {
     unoptimized: true,
-  }
+  },
+  webpack(config) {
```
Comment (benjagm) [inline next.config.js:9]: Why is this change required?

**Review by benjagm — CHANGES_REQUESTED** [review]
Added some comments.

**Comment by gregsdennis** [comment]
I would also like a chance to review this from a UX perspective.  I'm curious how the Bowtie integration looks. 

Still waiting for a good build so that a preview is generated.

File: data/tooling-data.yaml (hunk 1 above)
Comment (DarhkVoyd) [reply data/tooling-data.yaml:904]: The data model is implemented from the json schema eco-system tooling self identification schema, since it already had a landscape property with a logo field, I utilised the exact same model as I understood that the logo is more relevant to the landscape project. If you would like to instead extend the self identification schema and add logo as a root property, let me know.

File: next.config.js (hunk 2 above)
Comment (DarhkVoyd) [reply next.config.js:9]: This change is required to optimize how we manage and render icons across our website. Currently, we're fetching icons as images from the server at runtime, which leads to several inefficiencies:

1. **Unnecessary Network Requests**: Every icon is fetched as a separate image file, increasing the number of network requests. This impacts load times and server performance.
2. **Complex Theme Management**: To handle different themes, we're using multiple useEffect hooks to fetch and update icons based on the current theme. This approach is repetitive and adds unnecessary complexity to our code.

By switching to SVGR, we can embed SVGs directly as React components. This allows us to:

1. **Reduce Network Requests**: Icons are embedded at build time, eliminating the need to fetch them from the server during runtime.
2. **Simplify Theme Management**: We can easily manipulate icon styles natively with CSS, like using `dark:fill-black` to change the fill color, instead of relying on multiple useEffect hooks.

File: data/tooling-data.yaml (hunk 1 above)
Comment (DarhkVoyd) [reply data/tooling-data.yaml:904]: For now, the landscape property only exists in the schema and type interface but I only added a single logo to test it out. So, we can still update our data model slights to move the logo field to the root properties instead.

File: next.config.js (hunk 2 above)
Comment (DarhkVoyd) [reply next.config.js:9]: Such as #850, currently updating SVGs with theme would require `useTheme` and `useEffect` hooks, which again is repetitive and leads to inconsistencies sometimes. This would all be as simple as adding `dark:` classes to the svg components if we use svg's as react components and embed them during build time.

Comment (gregsdennis) [reply next.config.js:9]: Is this _necessary_ for this PR to work? If not, it should be in a separate change.

The effort is appreciated, but if it's unrelated it needs to be addressed as such.

Comment (DarhkVoyd) [reply next.config.js:9]: Kind of but I can change it! The theming with `useEffect` was a headache so I tried this as a fix. The icons in the PR depends on this but they are few so I can easily switch back. Lemme know!

Comment (gregsdennis) [reply next.config.js:9]: I'm okay if it's related.  I'll let @benjagm make the call.

Comment (DarhkVoyd) [reply next.config.js:9]: Sure! Just to be clear, this does not affect any existing svg bundling with webpack because all current svgs use public path of the deployed page to point to the resource and instead of javascript modules `import`.

File: data/tooling-data.yaml (hunk 1 above)
Comment (benjagm) [reply data/tooling-data.yaml:904]: Ok

File: next.config.js (hunk 2 above)
Comment (benjagm) [reply next.config.js:9]: Thanks everyone for the great discussions. Let's continue with the change here.

**Review by benjagm — APPROVED** [review]
I pushed some small changes to improve the raise issues/bowtie buttons.

Everything else looks great! Amazing job. It is great to see this working!

**Comment by DarhkVoyd** [comment]
@benjagm Thanks! I have rebase the PR with latest changes, fixed conflicts and linting issues. The current redirect takes us back to the previous implementations page, should I go ahead and remove it?

**Comment by benjagm** [comment]
> @benjagm Thanks! I have rebased the PR with latest changes, fixed conflicts and linting issues. The preview is not working as expected as the redirects take us back to the previous implementations page, should I go ahead and remove it?

Yes, We'll need to replace the old redirect available in the _redirects file:
Now we have:
`/tools /implementations 301`

We should replace it by:
`/implementations /tools 301`

**Comment by DarhkVoyd** [comment]
Now, we just need to add the backend API to complete the integration.

**Comment by benjagm** [comment]
> Still waiting for a good build so that a preview is generated.

Now we have the preview here: https://github.com/json-schema-org/website/pull/858#issuecomment-2294805144

**Comment by benjagm** [comment]
> Now, we just need to add the backend API to complete the integration.

Let's do it!

**Comment by DarhkVoyd** [comment]
I think we also need update the main nav link to `/tools`. Next router might not use the Cloudflare redirects.

**Comment by benjagm** [comment]
> I think we also need update the main nav link to `/tools`. Next router might not use the Cloudflare redirects.

Yes. Better to change that link. You can also perform a search in the codebase looking for internal links pointing to "/implementations" because I am sure there will be some more.

**Comment by gregsdennis** [comment]
What page is the change on?  The preview link just takes me to the home page.

**Comment by DarhkVoyd** [comment]
@gregsdennis I was just updating redirects and internal links, you can now easily preview changes on the tooling page.

**Comment by gregsdennis** [comment]
Is this Bowtie "integration" just a link to the Bowtie site?  From the description, I thought it'd be more.

**Comment by DarhkVoyd** [comment]
@gregsdennis to utilise the space on the tooling table most optimally the tooling table has a outlink to bowtie for anyone interested, If you click on any tool it will display the tool with detailed information including bowie report and compliance.

**Comment by DarhkVoyd** [comment]
I have realised that, currently we only fetch the latest bowtie report during the website build. So, if for some reason there is not a new build of the website for some time then we would still be displaying outdated report. Is this the expected behaviour or should I go ahead and add a workflow to instead keep an updated bowtie report in the data folder which runs on a regular interval such as every month?

**Review by benjagm — APPROVED** [review]
All good from my side. Thanks DV!!

**Comment by benjagm** [comment]
> Is this Bowtie "integration" just a link to the Bowtie site? From the description, I thought it'd be more.

When you click over a tool, you see the details in a pop-up and when that tool is integrated with Bowtie you'll see all the badges.

**Review by benjagm — APPROVED** [review]
This is ready from my side. Great job!!
